src/kibana/components/vislib/lib/legend.js

```diff
@@ -17,16 +21,19 @@ define(function (require) {
      * @param color {Function} Color function
      * @param _attr {Object|*} Reference to Vis options
      */
-    function Legend(vis, el, labels, color, _attr) {
+    function Legend(vis, data) {
       if (!(this instanceof Legend)) {
-        return new Legend(vis, el, labels, color, _attr);
+        return new Legend(vis, data);
       }
 
+      this.events = new Dispatch();
       this.vis = vis;
-      this.el = el;
-      this.labels = labels;
-      this.color = color;
-      this._attr = _.defaults(_attr || {}, {
+      this.el = vis.el;
+      this.data = this._getData(data);
```

Comment:
Can we use a different name here? You're passing in `data`, and `this._getData()` isn't just returning the value that was passed in... also, `this.data` isn't the same as the data argument... I can't keep this straight.